src/components/PeopleFilters.tsx

```diff
@@ -1,89 +1,146 @@
-export const PeopleFilters = () => {
+import classNames from 'classnames';
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { SearchLink } from './SearchLink';
+import { CENTURIES, SEX_FILTER_LINKS } from '../utils/constants';
+
+type Props = {
+  query: string,
+  sex: string,
+  centuries: string[],
+};
+
+export const PeopleFilters: React.FC<Props> = ({
+  query,
+  sex,
+  centuries,
+}) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  function handleChangeQuery(event: React.ChangeEvent<HTMLInputElement>) {
+    const params = new URLSearchParams(searchParams);
+
+    params.set('query', event.target.value);
+    setSearchParams(params);
+  }
+
+  function handleSexSelected(filterPath: string | null) {
+    const params = new URLSearchParams(searchParams);
+
+    params.set('sex', filterPath || '');
+    setSearchParams(params);
+  }
+
+  function toggleCenturies(cntr: string) {
+    const params = new URLSearchParams(searchParams);
+
+    const newCenturies = centuries.includes(cntr)
+      ? centuries.filter(prevCentury => prevCentury !== cntr)
+      : [...centuries, cntr];
+
+    params.delete('centuries');
+    newCenturies.forEach(cent => params.append('centuries', cent));
+    setSearchParams(params);
+  }
+
+  function handleClearAllCenturies() {
+    const params = new URLSearchParams(searchParams);
+
+    params.delete('centuries');
+    setSearchParams(params);
+  }
+
+  function handleResetAllFilters() {
+    const params = new URLSearchParams(searchParams);
+
+    params.delete('sex');
+    params.delete('query');
+    setSearchParams(params);
+  }
```

Comment:
You don't need these functions. You have `params` prop in your `SearchLink` for this